src/machine.rs

```diff
@@ -365,31 +387,75 @@ impl<'mir, 'tcx> Machine<'mir, 'tcx> for Evaluator<'tcx> {
         Ok(())
     }
 
+    fn access_local(
```

Comment:
I don't see this added to the machine trait in your rustc PR, what is happening here?

Reply:
Oh wait the trait already exists, right. But why is an implementation now needed?

Reply:
This is how I made [the test](https://github.com/rust-lang/rust/pull/70598#discussion_r403378561) that sends a pointer to a thread local to another thread to pass. Basically, the idea is to check whether the accessed local is a thread-local static and if yes, get the real allocation id in the same way as in `canonical_alloc_id`.